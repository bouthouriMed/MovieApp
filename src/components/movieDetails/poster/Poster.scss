.poster {
  flex: 1;
  max-width: 500px;

  img {
    width: 100%;
    border-radius: 12px;
    box-shadow: 0 6px 20px rgba(0, 0, 0, 0.7);
    transition: transform 0.3s;

    &:hover {
      transform: scale(1.05);
    }
  }

  @media (max-width: 768px) {
    max-width: 100%;
    img {
      border-radius: 10px;

      &:hover {
        transform: none;
      }
    }
  }
}
