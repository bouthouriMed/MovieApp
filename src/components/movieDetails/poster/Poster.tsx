import { memo } from "react";
import "./Poster.scss";

interface PosterProps {
  posterPath: string;
  title: string;
}

const Poster = ({ posterPath, title }: PosterProps) => {
  return (
    <div className="poster">
      <img src={`https://image.tmdb.org/t/p/w500${posterPath}`} alt={title} />
    </div>
  );
};

export default memo(Poster);
