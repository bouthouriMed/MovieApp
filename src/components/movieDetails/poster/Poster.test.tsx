import { render, screen } from "@testing-library/react";
import Poster from "./Poster";
import { describe, it, expect } from "vitest";
import { mockedMovie } from "@/__mocks__/test_mocks";

describe("Poster Component", () => {
  it("renders movie poster with correct alt text and src", () => {
    const { container } = render(
      <Poster posterPath={mockedMovie.poster_path} title={mockedMovie.title} />
    );

    const img = screen.getByAltText(mockedMovie.title);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute(
      "src",
      `https://image.tmdb.org/t/p/w500${mockedMovie.poster_path}`
    );

    expect(container).toMatchSnapshot();
  });
});
