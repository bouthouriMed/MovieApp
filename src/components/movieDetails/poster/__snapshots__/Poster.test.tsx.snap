// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Poster Component > renders movie poster with correct alt text and src 1`] = `
<div>
  <div
    class="poster"
  >
    <img
      alt="Test Movie"
      src="https://image.tmdb.org/t/p/w500/poster.jpg"
    />
  </div>
</div>
`;
